src/test/fuzz/orphanage.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <net_processing.h>
+#include <primitives/transaction.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <test/util/mining.h>
+#include <test/util/script.h>
+#include <test/util/setup_common.h>
+#include <txorphanage.h>
+#include <validation.h>
+
+
+FUZZ_TARGET(orphanage)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+
+    TxOrphanage orphanage;
+    std::set<uint256> orphan_work_set;
+    std::set<COutPoint> outpoints;
+    // initial outpoints used to construct transactions later
+    for (uint8_t i = 0; i < 4; i++) {
+        outpoints.emplace(uint256{i}, 0);
+    }
+
+    LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), 2 * DEFAULT_MAX_ORPHAN_TRANSACTIONS)
+    {
+        // construct transaction
+        const CTransactionRef tx = [&] {
+            CMutableTransaction tx_mut;
+            const auto num_in = fuzzed_data_provider.ConsumeIntegralInRange<uint32_t>(1, outpoints.size());
+            const auto num_out = fuzzed_data_provider.ConsumeIntegralInRange<uint32_t>(1, outpoints.size());
+            // pick unique outpoints from g_outpoints as input
+            for (uint32_t i = 0; i < num_in; i++) {
+                auto pop = outpoints.begin();
+                std::advance(pop, fuzzed_data_provider.ConsumeIntegralInRange<uint32_t>(0, outpoints.size() - 1));
+                const auto prevout = *pop;
+                tx_mut.vin.emplace_back(prevout);
+                outpoints.erase(pop);
+            }
+            // output amount will not affect txorphanage
+            for (uint32_t i = 0; i < num_out; i++) {
+                tx_mut.vout.emplace_back(CAmount{0}, P2WSH_OP_TRUE);
+            }
+            // restore previously removed outpoints
+            for (auto& in : tx_mut.vin) {
+                outpoints.insert(in.prevout);
+            }
+            const auto new_tx = MakeTransactionRef(tx_mut);
+            // add newly constructed transaction to outpoints
+            for (uint32_t i = 0; i < num_out; i++) {
+                outpoints.emplace(new_tx->GetHash(), i);
+            }
+            return new_tx;
+        }();
+
+        // trigger orphanage functions
+        LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), 2 * DEFAULT_MAX_ORPHAN_TRANSACTIONS)
```

Comment:
There are two limited loops in the file, the outer is determines how many transactions can be constructed, the inner (this loop you sugguested) determines how many times can `TxOrphange` methods be triggered. I am not sure the exact value to be set. I think it can be bigger, though the bool may limit most loops to have limited iteration time.